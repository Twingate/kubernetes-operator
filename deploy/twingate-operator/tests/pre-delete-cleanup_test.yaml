suite: Pre-delete Cleanup
templates:
  - pre-delete-cleanup.yaml
set:
  twingateOperator:
    network: <network slug>
    apiKey: <api key>
    remoteNetworkId: <remote network id>
tests:
  - it: should enable pre-delete cleanup job
    set:
      kubernetes-access-gateway:
        enabled: true
    asserts:
      - matchSnapshot: {}
