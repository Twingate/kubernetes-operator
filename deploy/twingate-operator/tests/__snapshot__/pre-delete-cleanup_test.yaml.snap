should enable pre-delete cleanup job:
  1: |
    apiVersion: batch/v1
    kind: Job
    metadata:
      annotations:
        helm.sh/hook: pre-delete
        helm.sh/hook-delete-policy: hook-succeeded
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: twingate-operator
        app.kubernetes.io/version: latest
        helm.sh/chart: twingate-operator-0.1.13
      name: RELEASE-NAME-twingate-operator-pre-delete-cleanup
      namespace: NAMESPACE
    spec:
      template:
        spec:
          containers:
            - command:
                - /bin/sh
                - -c
                - |
                  kubectl delete svc --cascade=foreground --timeout=30s --ignore-not-found -n NAMESPACE RELEASE-NAME-kubernetes-access-gateway
              image: rancher/kubectl:v1.34.0
              name: cleanup
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 1000
          restartPolicy: Never
          securityContext:
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-twingate-operator
